File: src/types/update.ts
export interface UpdateInfo {
  version: string;
  releaseNotes: string;
  mandatory: boolean;
  channel: string;
  assetName?: string | null;
  publishedAt?: string | null;
  downloaded: boolean;
}

export interface UpdateProgress {
  downloaded: number;
  total?: number | null;
  percent: number;
  assetName: string;
}

File: src-tauri/src/models/update.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateInfo {
    pub version: String,
    pub release_notes: String,
    pub mandatory: bool,
    pub channel: String,
    pub asset_name: Option<String>,
    pub published_at: Option<String>,
    pub downloaded: bool,
}
